drivers/temperature/adt7420/adt7420.c

```diff
@@ -250,3 +345,106 @@ float adt7420_get_temperature(struct adt7420_dev *dev)
 
 	return temp_c;
 }
+
+
+/**************************************************************************//**
+ * @brief Write to input shift register SPI interface.
+ *
+ * @param dev               - The device structure.
+ * @param register_address  - Command control bits.
+ * @param data              - Data to be written in input register.
+ *
+ * @return  read_back_data  - value read from register.
+******************************************************************************/
+uint16_t set_shift_reg(struct adt7420_dev *dev,
+		       uint8_t register_address,
+		       uint8_t *data)
+{
+	uint8_t data_buffer[3] = { 0, 0, 0 };
+	uint16_t read_back_data = 0;
+	uint8_t numBytes;
+
+	switch (register_address) {
+	case ADT7320_REG_STATUS:
+	case ADT7320_REG_CONFIG:
+	case ADT7320_REG_ID:
+	case ADT7320_REG_HIST:
+		numBytes = 2;
+		break;
+	default:
+		numBytes = 3;
+		break;
+	}
+
+	if (numBytes == 2) {
+		data_buffer[0] = data[0];
+		data_buffer[1] = data[1] & 0x00FF;
+		spi_write_and_read(dev->spi_desc, data_buffer, 2);
+		read_back_data = (data_buffer[0] & 0xFF00) | data_buffer[1];
+	} else {
+		data_buffer[0] = data[0];
+		data_buffer[1] = data[1] & 0x00FF;
+		data_buffer[2] = (data[2] & 0x00FF) << 8;
```

Comment:
If `data_buffer` is only represented on 8 bits here shifting something by 8 bits and assigning the value to `data_buffer` will truncate the result to probably 0, most of the time. Am I missing something here?

Reply:
nitpick: `data` points to 8-bit values. Masking with `0x00FF` seems pointless.

Reply:
data[2] is not used anywhere. Hence removed the line. There has been a slight modification made in the function definition to accommodate only the read back value calculation within the if condition